torch/distributed/_composable/replicate.py

```diff
@@ -68,6 +68,23 @@ def init(
             return
 
         self.has_initialized = True
+
+        device_mesh = kwargs.get("device_mesh", None)
+        if device_mesh is not None:
+            from torch.distributed.device_mesh import _mesh_resources
+            if _mesh_resources.get_parent_mesh(device_mesh) is not None:
```

Comment:
This is more for my understanding. Doesn't replicate init call into DDP init? Wondering why we need it in both.

Reply:
Because the way replicate use DDP is to create a `nn.ParameterList` and pass that new `module` to DDP. So even though DDP will call `_pre_dp_module_transform` over the passed module, that moudule's forward is never called.